pandas/core/indexes/interval.py

```diff
@@ -640,8 +645,18 @@ def mid(self):
             return Index(0.5 * (self.left.values + self.right.values))
         except TypeError:
             # datetime safe version
+            tz = self.right.tz
+            freq = self.right.freq
             delta = self.right.values - self.left.values
-            return Index(self.left.values + 0.5 * delta)
+
+            # handle tz aware
+            if tz:
+                data = self.left.tz_localize(None) + 0.5 * delta
+                data = data.tz_localize(tz)
```

Comment:
Is there a better way to do this computation?  Couldn't get the correct result without removing the time zone, then adding it back afterwards.

Trying to get the midpoint between elements in `left` and `right`.  Basic setup is along the lines of:
```python
In [2]: left = pd.date_range('2017-01-01', periods=3, tz='US/Eastern')
   ...: right = pd.date_range('2017-01-02', periods=3, tz='US/Eastern')
   ...: delta = right.values - left.values
   ...:
```
The existing method removed the tz, and had the wrong time (all times should be 12:00):
```python
In [3]: left.values + 0.5 * delta
Out[3]:
array(['2017-01-01T17:00:00.000000000', '2017-01-02T17:00:00.000000000',
       '2017-01-03T17:00:00.000000000'], dtype='datetime64[ns]')
```
Operating on the index itself instead of the values keeps the tz, but the time is still wrong:
```python
In [4]: left + 0.5 * delta
Out[4]:
DatetimeIndex(['2017-01-01 17:00:00-05:00', '2017-01-02 17:00:00-05:00',
               '2017-01-03 17:00:00-05:00'],
              dtype='datetime64[ns, US/Eastern]', freq='D')
```
Having `delta` as a `TimdeltaIndex` raises:
```python
In [20]: left + 0.5 * (right - left)
---------------------------------------------------------------------------
TypeError: data type not understood
```

Reply:
[20] should work, we might have an issue about this; let's fix this fix this first in another PR. don't use ``.values`` with a datetime generally, you either want to operate on the ``.asi8``, the actual values (generally you will also need to .tz_localize(None) first to make them UTC, then reverse this later though).

Reply:
Ah, yes, looks like #17558 is the issue for this, and #17583 is an open PR, albeit somewhat stale.

Reply:
ok I closed that other PR as its stale. So options are to: issue a new PR before this one to fix #17558, or to do hacky fix here with a TODO, or xfail the tests for this (for now).

Reply:
I'm planning to look into that issue over the next few days and open new PR to fix it prior to this one being merged.

Reply:
great!